File: spoj/LINEGAME.py
input=raw_input()
n=int(input)
input=raw_input()
f=input.split(' ')
cong=int(f[0])
tru=0
x=range(2,n+1)
for i in x: 
	a=int(f[i-1])
	if tru<cong-a:
		tru=cong-a
	if cong<tru+a:
		cong=tru+a 
else:print cong

File: spoj/LINEGAME.pas
Program LINEGAME;
var cong,tru:int64;
n:longint;
//-------------------
procedure Solve;
var i,a:longint;
begin
    readln(n);
    read(cong);
    tru:=0;
    for i:=2 to n do
          begin
          read(a);
             if tru<cong-a then tru:=cong-a;
             if cong<tru+a then cong:=tru+a;
          end;
    writeln(cong);
end;
//-------------------
begin
          Solve;

end.

